pysph/tools/ipy_viewer.py

```diff
@@ -410,16 +456,24 @@ def interactive_plot(self):
                         temp_data[array_name],
                         pa_widgets.scalar.value
                 )
-                c = c + abs(np.min(c))
-                cmap = pa_widgets.scalar_cmap.value
-                colormap = getattr(mpl.cm, cmap)
-                sct = self._scatters[array_name]
-                cmax = np.max(c)
-                if cmax != 0:
-                    sct.set_facecolors(colormap(c*1.0/cmax))
+                colormap = getattr(
+                    plt.cm,
+                    pa_widgets.scalar_cmap.value
+                )
+                arr = self._normalized_for_colormaps(c)
```

Comment:
The comments below are a sign that a refactor is in order, instead can't you use this:
```
min_c, max_c, abs_max_c, c_norm = self._normalized_for_colormaps(c)
```
This eliminates the need for the comments and is easier to read.  Later on also `arr[2]` doesn't convey much but the variable name will make the code easier to read.

Reply:
Agreed, this does improve readability a lot. I've changed this as per your suggestion in the latest commit.